man/rgbasm.5

```diff
@@ -328,76 +328,100 @@ with a non-zero constant as either operand will be constant 1, even if the other
+.Pp
+By default,
+.Nm
+uses the precision specified by the
 .Fl Q
-command-line option.
-You can also specify a precise fixed-point value by appending a
+command-line option and/or by
+.Ic OPT Q
+.Pq see Sx Changing options while assembling ;
+the default precision is 16 bits (meaning the lower 16 bits are used for the fractional part, so the numbers count in 65536ths).
+The precision of a literal can be overridden by appending a
 .Dq q
-to it followed by the number of fractional bits, such as
-.Ql 12.34q8 .
-.Pp
-Since fixed-point values are still just integers, you can use them in normal integer expressions.
-Some integer operators like
-.Sq +
-and
-.Sq -
-don't care whether the operands are integers or fixed-point.
-You can easily truncate a fixed-point number into an integer by shifting it right by 16 bits.
-It follows that you can convert an integer to a fixed-point number by shifting it left.
-.Pp
-The following functions are designed to operate with fixed-point numbers:
+followed by the number of fractional bits:
+.Ql 1234.5q8
+is equal to $0004d280
 .EQ
 delim $$
 .EN
+($= 1234.5 * 2 sup 8$).
+.Pp
+The following functions are designed to operate with fixed-point numbers:
 .Bl -column -offset indent "ATAN2(y, x)"
 .It Sy Name Ta Sy Operation
-.It Fn DIV x y Ta Fixed-point division $( x \[di] y ) \[mu] ( 2 ^ precision )$
-.It Fn MUL x y Ta Fixed-point multiplication $( x \[mu] y ) \[di] ( 2 ^ precision )$
-.It Fn FMOD x y Ta Fixed-point modulo $( x % y ) \[di] ( 2 ^ precision )$
-.It Fn POW x y Ta $x$ to the $y$ power
-.It Fn LOG x y Ta Logarithm of $x$ to the base $y$
+.It Fn DIV x y Ta Fixed-point division
+.It Fn MUL x y Ta Fixed-point multiplication
+.It Fn FMOD x y Ta Fixed-point modulo
+.It Fn POW x y Ta $x sup y$
+.It Fn LOG x y Ta $log sub y ( x )$ (such that
+.Ic LOG Ns ( Ic POW Ns ( x , y ) , y ) == x)
 .It Fn ROUND x Ta Round $x$ to the nearest integer
-.It Fn CEIL x Ta Round $x$ up to an integer
-.It Fn FLOOR x Ta Round $x$ down to an integer
+.It Fn CEIL x Ta Round $x$ up to the nearest integer
+.It Fn FLOOR x Ta Round $x$ down to the nearest integer
 .It Fn SIN x Ta Sine of $x$
 .It Fn COS x Ta Cosine of $x$
 .It Fn TAN x Ta Tangent of $x$
-.It Fn ASIN x Ta Inverse sine of $x$
-.It Fn ACOS x Ta Inverse cosine of $x$
-.It Fn ATAN x Ta Inverse tangent of $x$
+.It Fn ASIN x Ta Inverse sine of $x$ (such that
+.Ic ASIN Ns ( Ic SIN Ns ( x ) ) == x)
+.It Fn ACOS x Ta Inverse cosine of $x$ (such that
+.Ic ACOS Ns ( Ic COS Ns ( x ) ) == x)
+.It Fn ATAN x Ta Inverse tangent of $x$ (such that
+.Ic ATAN Ns ( Ic TAN Ns ( x ) ) == x)
 .It Fn ATAN2 y x Ta Angle between $( x , y )$ and $( 1 , 0 )$
 .El
 .EQ
 delim off
 .EN
 .Pp
-All of these fixed-point functions can take an optional final argument, which is the precision to use.
+Operators not in the list above, such as
+.Sq +
+and
+.Sq - ,
+work correctly with fixed-point operands as well (but
```

Comment:
I'm not sure this precisely explains what's going on. Other operands do "work" with whatever numbers you want, the results will just not be interpretable as either the integers or the fixed-point values you might naively expect. (Also the list above has no "operators", just fixed-point functions.)

How about something like:

> There are no functions for fixed-point addition and subtraction because the '`+`' and '`-`' operators already do that when both operands are fixed-point.

> Note that some operators or functions are meaningful when combining integers and fixed-point values. For example, `2.0 * 3` is equivalent to `MUL(2.0, 3.0)`, and `6.0 / 2` is equivalent to `DIV(6.0, 2.0)`. Be careful and think about what the operations mean when doing this sort of thing.

Reply:
I wholeheartedly agree with the second block, but for the first one, I wonder if there are any operators we're missing?

(Additionally, shouldn't `%` work just as well as `FMOD`?)

Reply:
I don't see anything in the [operators table](https://rgbds.gbdev.io/docs/master/rgbasm.5#Operators) that ought to have a fixed-point function.

I'm not sure if `FMOD` behaves entirely the same as `%`. Maybe? Anyway, we already provide the function.

Reply:
%

Reply:
`FMOD` and `%` should be functionally identical. Quick test: `FMOD(x, 1.0)` will keep the fractional bits, and so will `x % 1.0`.

Reply:
Even with non-integral or negative operands, on either side?

Reply:
I don't even know what `%` does with negatives. Is that documented? Does it just follow C here?
Non-integral operands should work just fine.

Reply:
Yes, we document it to do the opposite of C:

> `%` is used to get the remainder of the corresponding division, so that `a / b * b + a % b == a` is always true. The result has the same sign as the divisor. This makes `a % b` equal to `(a + b) % b` or `(a - b) % b`.

`FMOD()` works like IEEE-754 `fmod()`.

Reply:
In that case it's probably slightly different for negatives. I don't remember which one of the two common definitions IEEE uses, but I'm pretty sure it's not that one.

Reply:
Sounds good to me! (It had better, since I wrote those comments. :P )

Reply:
Applied that change.